x/subspaces/keeper/msg_server.go

```diff
@@ -264,6 +485,65 @@ func (k msgServer) EditUserGroup(goCtx context.Context, msg *types.MsgEditUserGr
 	return &types.MsgEditUserGroupResponse{}, nil
 }
 
+// MoveUserGroup defines a rpc method for MsgMoveUserGroup
+func (k msgServer) MoveUserGroup(goCtx context.Context, msg *types.MsgMoveUserGroup) (*types.MsgMoveUserGroupResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// Check if the subspace exists
+	subspace, found := k.GetSubspace(ctx, msg.SubspaceID)
+	if !found {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "group with id %d not found", msg.SubspaceID)
+	}
+
+	// Check if the destination section exists
+	if !k.HasSection(ctx, msg.SubspaceID, msg.NewSectionID) {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "section with id %d not found inside subspace %d", msg.NewSectionID, msg.SubspaceID)
+	}
+
+	// Check if the group exists
+	group, found := k.GetUserGroup(ctx, msg.SubspaceID, msg.GroupID)
+	if !found {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "group with id %d not found", msg.GroupID)
+	}
+
+	signer, err := sdk.AccAddressFromBech32(msg.Signer)
+	if err != nil {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid signer address: %s", msg.Signer)
+	}
+
+	// Check the permission to create a group
+	if !k.HasPermission(ctx, group.SubspaceID, group.SectionID, signer.String(), types.PermissionManageGroups) {
+		return nil, sdkerrors.Wrap(types.ErrPermissionDenied, "you cannot manage user groups in this subspace")
+	}
+
+	// Make sure that the user is not part of the group they want to change the permissions for, unless they are the owner
+	if subspace.Owner != msg.Signer && k.IsMemberOfGroup(ctx, msg.SubspaceID, msg.GroupID, signer.String()) {
```

Comment:
In the following case, the moderator who has the `ManageGroups` in A section can move the B section group to the root.
```
   Root
    |
    A
    |
    B
```
Instead of this check, It can be solved by `k.HasPermission(ctx, group.SubspaceID, msg.NewSectionID, signer.String(), types.PermissionManageGroups)`, to check permission in the new section.

Reply:
Updated the checks to include both `PermissionManageGroups` and `PermissionSetUserPermissions` for the destination section :+1: